<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.551">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jacob Fredegaard Hansen">
<meta name="dcterms.date" content="2024-10-01">
<meta name="description" content="Outline for clinical proteomics">

<title>Clinical Proteomics - Breast Cancer Proteomics Module</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="stylesheets/neoteroi-mkdocs.css">
<link rel="stylesheet" href="stylesheets/extra.css">
</head>

<body>

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#preliminary-work" id="toc-preliminary-work" class="nav-link active" data-scroll-target="#preliminary-work">Preliminary work</a>
  <ul class="collapse">
  <li><a href="#questions-for-understanding-the-paper" id="toc-questions-for-understanding-the-paper" class="nav-link" data-scroll-target="#questions-for-understanding-the-paper">Questions for Understanding the Paper</a></li>
  <li><a href="#supplementary-data" id="toc-supplementary-data" class="nav-link" data-scroll-target="#supplementary-data">Supplementary Data</a></li>
  <li><a href="#question-4-what-is-in-tmt131" id="toc-question-4-what-is-in-tmt131" class="nav-link" data-scroll-target="#question-4-what-is-in-tmt131">Question 4: What is in TMT131?</a></li>
  <li><a href="#question-5-what-is-the-purpose-of-having-such-a-sample" id="toc-question-5-what-is-the-purpose-of-having-such-a-sample" class="nav-link" data-scroll-target="#question-5-what-is-the-purpose-of-having-such-a-sample">Question 5: What is the purpose of having such a sample?</a></li>
  </ul></li>
  <li><a href="#analysis-of-ms-data-using-fragpipe" id="toc-analysis-of-ms-data-using-fragpipe" class="nav-link" data-scroll-target="#analysis-of-ms-data-using-fragpipe">Analysis of MS Data Using FragPipe</a>
  <ul class="collapse">
  <li><a href="#question-6-where-is-the-data-located" id="toc-question-6-where-is-the-data-located" class="nav-link" data-scroll-target="#question-6-where-is-the-data-located">Question 6: Where is the data located?</a></li>
  <li><a href="#question-7-what-is-the-proteomexchange-database" id="toc-question-7-what-is-the-proteomexchange-database" class="nav-link" data-scroll-target="#question-7-what-is-the-proteomexchange-database">Question 7: What is the ProteomeXChange database?</a></li>
  <li><a href="#question-8-what-is-the-accession-code-used-for-the-data-deposited-in-proteomexchange" id="toc-question-8-what-is-the-accession-code-used-for-the-data-deposited-in-proteomexchange" class="nav-link" data-scroll-target="#question-8-what-is-the-accession-code-used-for-the-data-deposited-in-proteomexchange">Question 8: What is the accession code used for the data deposited in ProteomeXChange?</a></li>
  <li><a href="#question-9-what-is-ftp-and-how-does-it-work" id="toc-question-9-what-is-ftp-and-how-does-it-work" class="nav-link" data-scroll-target="#question-9-what-is-ftp-and-how-does-it-work">Question 9: What is FTP, and how does it work?</a></li>
  <li><a href="#question-10-what-is-fragpipe-and-what-can-it-be-used-for" id="toc-question-10-what-is-fragpipe-and-what-can-it-be-used-for" class="nav-link" data-scroll-target="#question-10-what-is-fragpipe-and-what-can-it-be-used-for">Question 10: What is FragPipe, and what can it be used for?</a></li>
  <li><a href="#question-11-if-you-were-not-using-fragpipe-for-this-part-of-the-teaching-module-which-other-software-tools-could-be-used-instead-please-include-a-few-examples." id="toc-question-11-if-you-were-not-using-fragpipe-for-this-part-of-the-teaching-module-which-other-software-tools-could-be-used-instead-please-include-a-few-examples." class="nav-link" data-scroll-target="#question-11-if-you-were-not-using-fragpipe-for-this-part-of-the-teaching-module-which-other-software-tools-could-be-used-instead-please-include-a-few-examples.">Question 11: If you were not using FragPipe for this part of the teaching module, which other software tools could be used instead? Please include a few examples.</a></li>
  <li><a href="#question-12-what-are-the-advantages-of-using-fragpipe" id="toc-question-12-what-are-the-advantages-of-using-fragpipe" class="nav-link" data-scroll-target="#question-12-what-are-the-advantages-of-using-fragpipe">Question 12: What are the advantages of using FragPipe?</a></li>
  </ul></li>
  <li><a href="#download-data-from-the-paper" id="toc-download-data-from-the-paper" class="nav-link" data-scroll-target="#download-data-from-the-paper">Download Data from the Paper</a>
  <ul class="collapse">
  <li><a href="#question-13-what-is-the-code-above-doing-please-describe-the-code-and-what-it-is-used-for." id="toc-question-13-what-is-the-code-above-doing-please-describe-the-code-and-what-it-is-used-for." class="nav-link" data-scroll-target="#question-13-what-is-the-code-above-doing-please-describe-the-code-and-what-it-is-used-for.">Question 13: What is the code above doing? Please describe the code and what it is used for.</a></li>
  <li><a href="#question-14-please-find-the-address." id="toc-question-14-please-find-the-address." class="nav-link" data-scroll-target="#question-14-please-find-the-address.">Question 14: Please find the address.</a></li>
  <li><a href="#question-15-now-we-have-access-to-the-data-deposited-on-the-ftp-server.-briefly-describe-the-contents-of-the-folder-in-the-ftp-server." id="toc-question-15-now-we-have-access-to-the-data-deposited-on-the-ftp-server.-briefly-describe-the-contents-of-the-folder-in-the-ftp-server." class="nav-link" data-scroll-target="#question-15-now-we-have-access-to-the-data-deposited-on-the-ftp-server.-briefly-describe-the-contents-of-the-folder-in-the-ftp-server.">Question 15: Now, we have access to the data deposited on the FTP-server. Briefly describe the contents of the folder in the FTP-server.</a></li>
  <li><a href="#question-16-please-describe-what-the-code-is-doing-by-explaining-the-functions-used." id="toc-question-16-please-describe-what-the-code-is-doing-by-explaining-the-functions-used." class="nav-link" data-scroll-target="#question-16-please-describe-what-the-code-is-doing-by-explaining-the-functions-used.">Question 16: Please describe what the code is doing by explaining the functions used.</a></li>
  <li><a href="#getting-started-with-fragpipe" id="toc-getting-started-with-fragpipe" class="nav-link" data-scroll-target="#getting-started-with-fragpipe">Getting started with FragPipe</a></li>
  <li><a href="#question-17-which-workflow-should-you-select-hint-how-many-tmt-tags-are-given-in-the-table-in-supplementary-data-1." id="toc-question-17-which-workflow-should-you-select-hint-how-many-tmt-tags-are-given-in-the-table-in-supplementary-data-1." class="nav-link" data-scroll-target="#question-17-which-workflow-should-you-select-hint-how-many-tmt-tags-are-given-in-the-table-in-supplementary-data-1.">Question 17: Which workflow should you select? HINT: How many TMT-tags are given in the table in Supplementary Data 1.</a></li>
  <li><a href="#question-18-what-is-the-database-file-used-for-in-fragpipe-and-why-is-it-important" id="toc-question-18-what-is-the-database-file-used-for-in-fragpipe-and-why-is-it-important" class="nav-link" data-scroll-target="#question-18-what-is-the-database-file-used-for-in-fragpipe-and-why-is-it-important">Question 18: What is the database file used for in FragPipe and why is it important?</a></li>
  <li><a href="#question-19-which-organism-should-you-select-for-downloading-the-database-file" id="toc-question-19-which-organism-should-you-select-for-downloading-the-database-file" class="nav-link" data-scroll-target="#question-19-which-organism-should-you-select-for-downloading-the-database-file">Question 19: Which organism should you select for downloading the database file?</a></li>
  <li><a href="#question-20-describe-the-relation-between-decoys-and-fdr-by-answering-these-questions" id="toc-question-20-describe-the-relation-between-decoys-and-fdr-by-answering-these-questions" class="nav-link" data-scroll-target="#question-20-describe-the-relation-between-decoys-and-fdr-by-answering-these-questions">Question 20: Describe the relation between decoys and FDR by answering these questions:</a></li>
  <li><a href="#question-21-what-is-msfragger" id="toc-question-21-what-is-msfragger" class="nav-link" data-scroll-target="#question-21-what-is-msfragger">Question 21: What is MSFragger?</a></li>
  <li><a href="#question-22-how-does-msfragger-work" id="toc-question-22-how-does-msfragger-work" class="nav-link" data-scroll-target="#question-22-how-does-msfragger-work">Question 22: How does MSFragger work?</a></li>
  <li><a href="#question-23-why-is-it-important-to-run-msfragger-for-this-analysis" id="toc-question-23-why-is-it-important-to-run-msfragger-for-this-analysis" class="nav-link" data-scroll-target="#question-23-why-is-it-important-to-run-msfragger-for-this-analysis">Question 23: Why is it important to run MSFragger for this analysis?</a></li>
  <li><a href="#question-24-what-do-you-expect-from-the-output-results-consider-the-number-of-files-you-have-provided-for-this-search." id="toc-question-24-what-do-you-expect-from-the-output-results-consider-the-number-of-files-you-have-provided-for-this-search." class="nav-link" data-scroll-target="#question-24-what-do-you-expect-from-the-output-results-consider-the-number-of-files-you-have-provided-for-this-search.">Question 24: What do you expect from the output results? Consider the number of files you have provided for this search.</a></li>
  <li><a href="#question-25-can-we-use-this-output-based-on-the-limited-number-of-sample-files-for-the-downstream-analysis-whywhy-not" id="toc-question-25-can-we-use-this-output-based-on-the-limited-number-of-sample-files-for-the-downstream-analysis-whywhy-not" class="nav-link" data-scroll-target="#question-25-can-we-use-this-output-based-on-the-limited-number-of-sample-files-for-the-downstream-analysis-whywhy-not">Question 25: Can we use this output based on the limited number of sample files for the downstream analysis? Why/Why not?</a></li>
  <li><a href="#question-26-what-does-it-mean-that-the-sample-tissues-have-been-fractionated-according-to-paper-please-describe-that-process-and-study-design.-will-more-fractions-lead-to-better-proteome-coverage" id="toc-question-26-what-does-it-mean-that-the-sample-tissues-have-been-fractionated-according-to-paper-please-describe-that-process-and-study-design.-will-more-fractions-lead-to-better-proteome-coverage" class="nav-link" data-scroll-target="#question-26-what-does-it-mean-that-the-sample-tissues-have-been-fractionated-according-to-paper-please-describe-that-process-and-study-design.-will-more-fractions-lead-to-better-proteome-coverage">Question 26: What does it mean that the sample tissues have been fractionated according to <em>PAPER</em>? Please describe that process and study design. Will more fractions lead to better proteome coverage?</a></li>
  <li><a href="#question-27-what-kind-of-output-does-fragpipe-provide" id="toc-question-27-what-kind-of-output-does-fragpipe-provide" class="nav-link" data-scroll-target="#question-27-what-kind-of-output-does-fragpipe-provide">Question 27: What kind of output does FragPipe provide?</a></li>
  </ul></li>
  <li><a href="#further-interpretation-and-analysis-of-fragpipe-results" id="toc-further-interpretation-and-analysis-of-fragpipe-results" class="nav-link" data-scroll-target="#further-interpretation-and-analysis-of-fragpipe-results">Further Interpretation and Analysis of FragPipe Results</a>
  <ul class="collapse">
  <li><a href="#question-28-give-a-brief-description-of-the-contents-of-the-table.-what-does-the-table-contain-across-the-rows-and-the-columns" id="toc-question-28-give-a-brief-description-of-the-contents-of-the-table.-what-does-the-table-contain-across-the-rows-and-the-columns" class="nav-link" data-scroll-target="#question-28-give-a-brief-description-of-the-contents-of-the-table.-what-does-the-table-contain-across-the-rows-and-the-columns">Question 28: Give a brief description of the contents of the table. What does the table contain across the rows and the columns?</a></li>
  <li><a href="#reading-the-data" id="toc-reading-the-data" class="nav-link" data-scroll-target="#reading-the-data">Reading the Data</a></li>
  <li><a href="#visualization-of-the-data" id="toc-visualization-of-the-data" class="nav-link" data-scroll-target="#visualization-of-the-data">Visualization of the Data</a></li>
  <li><a href="#dynamic-range-similarity-and-missing-values" id="toc-dynamic-range-similarity-and-missing-values" class="nav-link" data-scroll-target="#dynamic-range-similarity-and-missing-values">Dynamic Range, Similarity, and Missing Values</a></li>
  <li><a href="#principal-component-analysis" id="toc-principal-component-analysis" class="nav-link" data-scroll-target="#principal-component-analysis">Principal Component Analysis</a></li>
  <li><a href="#hierarchical-clustering" id="toc-hierarchical-clustering" class="nav-link" data-scroll-target="#hierarchical-clustering">Hierarchical Clustering</a></li>
  <li><a href="#digging-deeper-into-the-behavior-of-one-protein" id="toc-digging-deeper-into-the-behavior-of-one-protein" class="nav-link" data-scroll-target="#digging-deeper-into-the-behavior-of-one-protein">Digging Deeper into the Behavior of One Protein</a></li>
  <li><a href="#statistics-and-clustering-using-vsclust" id="toc-statistics-and-clustering-using-vsclust" class="nav-link" data-scroll-target="#statistics-and-clustering-using-vsclust">Statistics and Clustering using VSClust</a></li>
  <li><a href="#go-terms-and-pathways" id="toc-go-terms-and-pathways" class="nav-link" data-scroll-target="#go-terms-and-pathways">GO Terms and Pathways</a></li>
  </ul></li>
  <li><a href="#personal-details" id="toc-personal-details" class="nav-link" data-scroll-target="#personal-details">Personal Details</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Breast Cancer Proteomics Module</h1>
</div>

<div>
  <div class="description">
    Outline for clinical proteomics
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Jacob Fredegaard Hansen </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 1, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<!--
# Put above to hide navigation (left), toc (right) or footer (bottom)

hide:
  - navigation 
  - toc
  - footer 

# You should hide the navigation if there are no subsections
# You should hide the Table of Contents if there are no important titles
-->
<p>!!! note “Section Overview”</p>
<pre><code>&amp;#128368; **Total Time Estimation:** 8 hours

&amp;#128172; **Learning Objectives:**    
    1. Understand the workflow of the study. &lt;br&gt;
    2. Retrieve and understand the study design and data from selected papers. &lt;br&gt;
    3. Download and preprocess proteomics data using FragPipe. &lt;br&gt;
    4. Set up and use FragPipe for TMT-labeled MS data analysis. &lt;br&gt;
    5. Create accurate annotation files for data analysis.</code></pre>
<section id="preliminary-work" class="level2">
<h2 class="anchored" data-anchor-id="preliminary-work">Preliminary work</h2>
<p>For this work, we will use the data that was used and analyzed in the paper <em><a href="https://www-nature-com.proxy1-bib.sdu.dk/articles/s41467-019-09018-y">Breast cancer quantitative proteome and proteogenomic landscape</a></em> by Johansson <em>et al.</em>, which compares subgroups of breast cancer tumors from the Oslo2 cohort.</p>
<p>Before delving into the actual analysis of the data in FragPipe, we must initially:</p>
<ol type="1">
<li>Read and understand the study design of the paper.</li>
<li>Understand the data being used from the paper.</li>
</ol>
<section id="questions-for-understanding-the-paper" class="level3">
<h3 class="anchored" data-anchor-id="questions-for-understanding-the-paper">Questions for Understanding the Paper</h3>
<p>To better understand the paper, we have formulated some questions that should help clarify the study design, aim, and overall scope of the work. These questions are listed below:</p>
<ul>
<li>…</li>
<li>…</li>
<li>…</li>
</ul>
</section>
<section id="supplementary-data" class="level3">
<h3 class="anchored" data-anchor-id="supplementary-data">Supplementary Data</h3>
<p>Before we proceed and download the data available from the paper, we must first delve into some of the details in Supplementary Data 1. This is a large table containing the quantitative proteome data from the Oslo2 breast cancer cohort, which includes 45 subgroups of cancer tumors and relates to Figures 1-6 in the paper.</p>
<p>To better understand the supplementary data, we have prepared guiding questions to aid in interpreting the table. You can find the supplementary data in the paper <a href="https://www-nature-com.proxy1-bib.sdu.dk/articles/s41467-019-09018-y#Sec15">here</a>.</p>
<!-- Question 1 -->
<h3 class="question anchored">
Question 1: Give a short description of what the table contains.
</h3>
<textarea id="question1" placeholder="Type your answer here..." class="input-box"></textarea>
<!-- Question 2 -->
<h3 class="question anchored">
Question 2: What does the tumor ID indicate?
</h3>
<textarea id="question2" placeholder="Type your answer here..." class="input-box"></textarea>
<!-- Question 3 -->
<h3 class="question anchored">
Question 3: Provide a brief description of TMT-labelled mass spectrometry proteomics data and how it is experimentally executed.
</h3>
<textarea id="question3" placeholder="Type your answer here..." class="input-box"></textarea>
<section id="tmt10-tags-and-tumor-ids" class="level4">
<h4 class="anchored" data-anchor-id="tmt10-tags-and-tumor-ids">TMT10-Tags and Tumor IDs</h4>
<p>Fill in the table below by entering the corresponding TMT10-tags and Tumor IDs. Once submitted, you’ll get instant feedback on whether your input is correct.</p>
<form id="tmtForm">
<table style="width: 100%; border-collapse: collapse;">
<tbody><tr>
<th style="width: 30%;">
TMT/Plex Set
</th>
<th style="width: 14%;">
Set 1
</th>
<th style="width: 14%;">
Set 2
</th>
<th style="width: 14%;">
Set 3
</th>
<th style="width: 14%;">
Set 4
</th>
<th style="width: 14%;">
Set 5
</th>
</tr>
<tr>
<td>
TMT126
</td>
<td>
<input type="text" id="set1_126" name="set1_126" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_126" name="set2_126" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_126" name="set3_126" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_126" name="set4_126" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_126" name="set5_126" style="width: 100%;">
</td>
</tr>
<tr>
<td>
TMT127N
</td>
<td>
<input type="text" id="set1_127N" name="set1_127N" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_127N" name="set2_127N" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_127N" name="set3_127N" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_127N" name="set4_127N" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_127N" name="set5_127N" style="width: 100%;">
</td>
</tr>
<tr>
<td>
TMT127C
</td>
<td>
<input type="text" id="set1_127C" name="set1_127C" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_127C" name="set2_127C" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_127C" name="set3_127C" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_127C" name="set4_127C" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_127C" name="set5_127C" style="width: 100%;">
</td>
</tr>
<tr>
<td>
TMT128N
</td>
<td>
<input type="text" id="set1_128N" name="set1_128N" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_128N" name="set2_128N" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_128N" name="set3_128N" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_128N" name="set4_128N" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_128N" name="set5_128N" style="width: 100%;">
</td>
</tr>
<tr>
<td>
TMT128C
</td>
<td>
<input type="text" id="set1_128C" name="set1_128C" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_128C" name="set2_128C" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_128C" name="set3_128C" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_128C" name="set4_128C" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_128C" name="set5_128C" style="width: 100%;">
</td>
</tr>
<tr>
<td>
TMT129N
</td>
<td>
<input type="text" id="set1_129N" name="set1_129N" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_129N" name="set2_129N" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_129N" name="set3_129N" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_129N" name="set4_129N" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_129N" name="set5_129N" style="width: 100%;">
</td>
</tr>
<tr>
<td>
TMT129C
</td>
<td>
<input type="text" id="set1_129C" name="set1_129C" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_129C" name="set2_129C" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_129C" name="set3_129C" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_129C" name="set4_129C" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_129C" name="set5_129C" style="width: 100%;">
</td>
</tr>
<tr>
<td>
TMT130N
</td>
<td>
<input type="text" id="set1_130N" name="set1_130N" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_130N" name="set2_130N" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_130N" name="set3_130N" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_130N" name="set4_130N" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_130N" name="set5_130N" style="width: 100%;">
</td>
</tr>
<tr>
<td>
TMT130C
</td>
<td>
<input type="text" id="set1_130C" name="set1_130C" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_130C" name="set2_130C" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_130C" name="set3_130C" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_130C" name="set4_130C" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_130C" name="set5_130C" style="width: 100%;">
</td>
</tr>
<tr>
<td>
TMT131
</td>
<td>
<input type="text" id="set1_131" name="set1_131" style="width: 100%;">
</td>
<td>
<input type="text" id="set2_131" name="set2_131" style="width: 100%;">
</td>
<td>
<input type="text" id="set3_131" name="set3_131" style="width: 100%;">
</td>
<td>
<input type="text" id="set4_131" name="set4_131" style="width: 100%;">
</td>
<td>
<input type="text" id="set5_131" name="set5_131" style="width: 100%;">
</td>
</tr>
</tbody></table>
<button type="button" onclick="checkAnswers()">
Submit
</button>
</form>
<p id="feedback">
</p>
<script>
  // Define correct answers for validation
  const correctAnswers = {
    set1_126: 'OSL.53E', set2_126: 'OSL.4AF', set3_126: 'OSL.43C', set4_126: 'OSL.43A', set5_126: 'OSL.40A',
    set1_127N: 'OSL.567', set2_127N: 'OSL.46E', set3_127N: 'OSL.493', set4_127N: 'OSL.406', set5_127N: 'OSL.40E',
    set1_127C: 'OSL.3FF', set2_127C: 'OSL.494', set3_127C: 'OSL.4D9', set4_127C: 'OSL.47C', set5_127C: 'OSL.3FA',
    set1_128N: 'OSL.55F', set2_128N: 'OSL.457', set3_128N: 'OSL.56F', set4_128N: 'OSL.524', set5_128N: 'OSL.521',
    set1_128C: 'OSL.46A', set2_128C: 'OSL.48B', set3_128C: 'OSL.405', set4_128C: 'OSL.443', set5_128C: 'OSL.46D',
    set1_129N: 'OSL.4B0', set2_129N: 'OSL.4B4', set3_129N: 'OSL.49E', set4_129N: 'OSL.458', set5_129N: 'OSL.54D',
    set1_129C: 'OSL.4D6', set2_129C: 'OSL.449', set3_129C: 'OSL.441', set4_129C: 'OSL.53D', set5_129C: 'OSL.4BA',
    set1_130N: 'OSL.485', set2_130N: 'OSL.44E', set3_130N: 'OSL.430', set4_130N: 'OSL.540', set5_130N: 'OSL.579',
    set1_130C: 'OSL.41B', set2_130C: 'OSL.3EB', set3_130C: 'OSL.4FA', set4_130C: 'OSL.42E', set5_130C: 'OSL.57B',
    set1_131: 'Pool', set2_131: 'Pool', set3_131: 'Pool', set4_131: 'Pool', set5_131: 'Pool'
  };

  function checkAnswers() {
    for (const [id, correctAnswer] of Object.entries(correctAnswers)) {
      const input = document.getElementById(id);
      const userAnswer = input.value.trim();

      if (userAnswer === correctAnswer) {
        input.style.backgroundColor = 'lightgreen'; // Highlight in green for correct
      } else {
        input.style.backgroundColor = 'lightcoral'; // Highlight in red for incorrect
      }
    }
  }
</script>
</section>
</section>
<section id="question-4-what-is-in-tmt131" class="level3">
<h3 class="anchored" data-anchor-id="question-4-what-is-in-tmt131">Question 4: What is in TMT131?</h3>
<textarea id="question4" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-5-what-is-the-purpose-of-having-such-a-sample" class="level3">
<h3 class="anchored" data-anchor-id="question-5-what-is-the-purpose-of-having-such-a-sample">Question 5: What is the purpose of having such a sample?</h3>
<textarea id="question5" placeholder="Type your answer here..." class="input-box"></textarea>
<p>Now that we better understand the workflow of the study and the content of the data, we are ready to move on to the analysis performed by FragPipe in the next section.</p>
</section>
</section>
<section id="analysis-of-ms-data-using-fragpipe" class="level2">
<h2 class="anchored" data-anchor-id="analysis-of-ms-data-using-fragpipe">Analysis of MS Data Using FragPipe</h2>
<p>In this section of the teaching module, we will work with data from the paper. The first task is to download sample files from the paper, guided by the questions provided below:</p>
<section id="question-6-where-is-the-data-located" class="level3">
<h3 class="anchored" data-anchor-id="question-6-where-is-the-data-located">Question 6: Where is the data located?</h3>
<textarea id="question6" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-7-what-is-the-proteomexchange-database" class="level3">
<h3 class="anchored" data-anchor-id="question-7-what-is-the-proteomexchange-database">Question 7: What is the ProteomeXChange database?</h3>
<textarea id="question7" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-8-what-is-the-accession-code-used-for-the-data-deposited-in-proteomexchange" class="level3">
<h3 class="anchored" data-anchor-id="question-8-what-is-the-accession-code-used-for-the-data-deposited-in-proteomexchange">Question 8: What is the accession code used for the data deposited in ProteomeXChange?</h3>
<textarea id="question8" placeholder="Type your answer here..." class="input-box"></textarea>
<p>By examining the accession code for the data deposited on ProteomeXChange, we can access and download the data using FTP.</p>
</section>
<section id="question-9-what-is-ftp-and-how-does-it-work" class="level3">
<h3 class="anchored" data-anchor-id="question-9-what-is-ftp-and-how-does-it-work">Question 9: What is FTP, and how does it work?</h3>
<textarea id="question9" placeholder="Type your answer here..." class="input-box"></textarea>
<p>For downloading the data, we will use the <strong>Proteomics Sandbox Application</strong> on UCloud. This platform allows us to access the necessary storage capacity as well as the computational power required to execute this process.</p>
<p>The <strong>Proteomics Sandbox Application</strong> is a virtual environment that includes multiple software tools, including <strong>FragPipe</strong> for analyzing proteomics data.</p>
<p>You can find the <strong>Proteomics Sandbox Application</strong> on UCloud <a href="https://cloud.sdu.dk/app/jobs/create?app=proteomics">here</a>.</p>
<p>First, we will download the data for the sample files to be used in <strong>FragPipe</strong>. Then, we will launch <strong>FragPipe</strong> to run the first analysis of the data. Before doing so, we have some questions regarding <strong>FragPipe</strong> and its usability:</p>
</section>
<section id="question-10-what-is-fragpipe-and-what-can-it-be-used-for" class="level3">
<h3 class="anchored" data-anchor-id="question-10-what-is-fragpipe-and-what-can-it-be-used-for">Question 10: What is FragPipe, and what can it be used for?</h3>
<textarea id="question10" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-11-if-you-were-not-using-fragpipe-for-this-part-of-the-teaching-module-which-other-software-tools-could-be-used-instead-please-include-a-few-examples." class="level3">
<h3 class="anchored" data-anchor-id="question-11-if-you-were-not-using-fragpipe-for-this-part-of-the-teaching-module-which-other-software-tools-could-be-used-instead-please-include-a-few-examples.">Question 11: If you were not using FragPipe for this part of the teaching module, which other software tools could be used instead? Please include a few examples.</h3>
<textarea id="question11" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-12-what-are-the-advantages-of-using-fragpipe" class="level3">
<h3 class="anchored" data-anchor-id="question-12-what-are-the-advantages-of-using-fragpipe">Question 12: What are the advantages of using FragPipe?</h3>
<textarea id="question12" placeholder="Type your answer here..." class="input-box"></textarea>
<p>Simple analyses in <strong>FragPipe</strong> may only require 8 GB of RAM, while large-scale or complex analyses may require 24 GB of memory or more (<a href="https://fragpipe.nesvilab.org/docs/tutorial_fragpipe.html#:~:text=FragPipe%20runs%20on%20Windows%20and,24%20GB%20memory%20or%20more.">FragPipe Documentation</a>), which is why we will allocate 24 GB for this exercise.</p>
<p>In UCloud, the settings should look like this:</p>
<p><strong>SCREENSHOT HERE</strong></p>
<p>Before submitting the job, it is also recommended to create a personal folder where you can store both the data and the results generated by <strong>FragPipe</strong>. You can follow the step-by-step guide below:</p>
<p><strong>SCREENSHOT HERE</strong></p>
<p><strong>CAUTION!!!</strong></p>
<p>Make sure to allocate the right number of hours before submitting the job. If the time runs out, the job will be canceled, and all progress will be lost. However, you can always extend the job duration if more time is required after submission.</p>
<p>Time can pass quickly when working, so we recommend initially allocating 2 hours for the job. Now, we are ready to submit the job and launch the virtual environment of the <strong>Proteomics Sandbox Application</strong>.</p>
</section>
</section>
<section id="download-data-from-the-paper" class="level2">
<h2 class="anchored" data-anchor-id="download-data-from-the-paper">Download Data from the Paper</h2>
<p>Initially, we will need to download the paper’s data. For this exercise, we will only use one sample file from each Plex Set/Pool.</p>
<p>We will use the terminal in the virtual environment for downloading the data. First, we need to update and download the necessary packages. You can do that by typing the following code:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sudo</span> apt-get update</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sudo</span> apt-get install lftp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="question-13-what-is-the-code-above-doing-please-describe-the-code-and-what-it-is-used-for." class="level3">
<h3 class="anchored" data-anchor-id="question-13-what-is-the-code-above-doing-please-describe-the-code-and-what-it-is-used-for.">Question 13: What is the code above doing? Please describe the code and what it is used for.</h3>
<textarea id="question13" placeholder="Type your answer here..." class="input-box"></textarea>
<p>Now, we can access the FTP server where the data is located. You will need the server address from the correct FTP-server, which can be found on the site for the accession code XXX in ProteomeXchange, previously visited. At the bottom of the page, you will find the FTP-server address where the data is stored.</p>
</section>
<section id="question-14-please-find-the-address." class="level3">
<h3 class="anchored" data-anchor-id="question-14-please-find-the-address.">Question 14: Please find the address.</h3>
<textarea id="question14" placeholder="Type your answer here..." class="input-box"></textarea>
<p>The address is used for accessing the data used in the study. To do so, we can use the package lftp that we just installed to access the server using the following code:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">lftp</span> [insert the address of the FTP server here]</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">lftp</span> ftp://ftp.......</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="question-15-now-we-have-access-to-the-data-deposited-on-the-ftp-server.-briefly-describe-the-contents-of-the-folder-in-the-ftp-server." class="level3">
<h3 class="anchored" data-anchor-id="question-15-now-we-have-access-to-the-data-deposited-on-the-ftp-server.-briefly-describe-the-contents-of-the-folder-in-the-ftp-server.">Question 15: Now, we have access to the data deposited on the FTP-server. Briefly describe the contents of the folder in the FTP-server.</h3>
<textarea id="question15" placeholder="Type your answer here..." class="input-box"></textarea>
<p>To download one sample file from each of the Plex Sets, you can use the following code in the terminal:</p>
<section id="code-here" class="level5">
<h5 class="anchored" data-anchor-id="code-here">CODE HERE</h5>
</section>
</section>
<section id="question-16-please-describe-what-the-code-is-doing-by-explaining-the-functions-used." class="level3">
<h3 class="anchored" data-anchor-id="question-16-please-describe-what-the-code-is-doing-by-explaining-the-functions-used.">Question 16: Please describe what the code is doing by explaining the functions used.</h3>
<textarea id="question16" placeholder="Type your answer here..." class="input-box"></textarea>
<p>If you added your own private folder to the UCloud session, you could now move the data into that folder for better management of the data you’re working with.</p>
<p>Next, we can launch FragPipe, which is located on the desktop. In this tutorial, we are using FragPipe version XX.YY in the June 2024 version of the Proteomics Sandbox Application.</p>
<p>Now that we have launched FragPipe, we need to configure the settings prior to running the analysis. Therefore, we have provided some guiding questions to help you set up the settings in FragPipe:</p>
</section>
<section id="getting-started-with-fragpipe" class="level3">
<h3 class="anchored" data-anchor-id="getting-started-with-fragpipe">Getting started with FragPipe</h3>
<p>Go to the “Workflow” tab to set up the workflow for the analysis and import the data you have just downloaded.</p>
</section>
<section id="question-17-which-workflow-should-you-select-hint-how-many-tmt-tags-are-given-in-the-table-in-supplementary-data-1." class="level3">
<h3 class="anchored" data-anchor-id="question-17-which-workflow-should-you-select-hint-how-many-tmt-tags-are-given-in-the-table-in-supplementary-data-1.">Question 17: Which workflow should you select? HINT: How many TMT-tags are given in the table in Supplementary Data 1.</h3>
<textarea id="question17" placeholder="Type your answer here..." class="input-box"></textarea>
<p>Click ‘Load workflow’ after you have found and selected the correct workflow to be used.</p>
<p>Next, add your files by clicking on “Add files” and locate them in the designated folder for your raw files that you previously created.</p>
<p>Now you should relocate to the “Database” tab. Here you can either download or browse for an already preexisting database file. In this case, we will simply download the latest database file.</p>
</section>
<section id="question-18-what-is-the-database-file-used-for-in-fragpipe-and-why-is-it-important" class="level3">
<h3 class="anchored" data-anchor-id="question-18-what-is-the-database-file-used-for-in-fragpipe-and-why-is-it-important">Question 18: What is the database file used for in FragPipe and why is it important?</h3>
<textarea id="question18" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-19-which-organism-should-you-select-for-downloading-the-database-file" class="level3">
<h3 class="anchored" data-anchor-id="question-19-which-organism-should-you-select-for-downloading-the-database-file">Question 19: Which organism should you select for downloading the database file?</h3>
<textarea id="question19" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-20-describe-the-relation-between-decoys-and-fdr-by-answering-these-questions" class="level3">
<h3 class="anchored" data-anchor-id="question-20-describe-the-relation-between-decoys-and-fdr-by-answering-these-questions">Question 20: Describe the relation between decoys and FDR by answering these questions:</h3>
<ul>
<li>What are decoys?</li>
<li>Why should you add decoys?</li>
<li>How do decoys play a role in the FDR estimate?</li>
</ul>
<textarea id="question20" placeholder="Type your answer here..." class="input-box"></textarea>
<p>Next, you can go to the MSFragger tab to adjust the parameter settings for the search and matching of the theoretical and experimental peptide spectra.</p>
<p>Most of the settings used for MSFragger can be obtained from the paper <em>NAME OF PAPER</em>, which is referred to in the Methods and Materials section.</p>
<p>When all settings have been obtained, MSFragger should look something like this:</p>
</section>
<section id="question-21-what-is-msfragger" class="level3">
<h3 class="anchored" data-anchor-id="question-21-what-is-msfragger">Question 21: What is MSFragger?</h3>
<textarea id="question21" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-22-how-does-msfragger-work" class="level3">
<h3 class="anchored" data-anchor-id="question-22-how-does-msfragger-work">Question 22: How does MSFragger work?</h3>
<textarea id="question22" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-23-why-is-it-important-to-run-msfragger-for-this-analysis" class="level3">
<h3 class="anchored" data-anchor-id="question-23-why-is-it-important-to-run-msfragger-for-this-analysis">Question 23: Why is it important to run MSFragger for this analysis?</h3>
<textarea id="question23" placeholder="Type your answer here..." class="input-box"></textarea>
<p>Finally, we can navigate to the “Run” tab and run the analysis. For that, we must choose an output directory for the results of the search made by FragPipe. Once you have adjusted that, you are ready to click on “Run”.</p>
<p>This process might take some time, so make sure that you still have enough hours allocated on your job on UCloud—otherwise, it will get terminated. Meanwhile, you can answer these questions:</p>
</section>
<section id="question-24-what-do-you-expect-from-the-output-results-consider-the-number-of-files-you-have-provided-for-this-search." class="level3">
<h3 class="anchored" data-anchor-id="question-24-what-do-you-expect-from-the-output-results-consider-the-number-of-files-you-have-provided-for-this-search.">Question 24: What do you expect from the output results? Consider the number of files you have provided for this search.</h3>
<textarea id="question24" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-25-can-we-use-this-output-based-on-the-limited-number-of-sample-files-for-the-downstream-analysis-whywhy-not" class="level3">
<h3 class="anchored" data-anchor-id="question-25-can-we-use-this-output-based-on-the-limited-number-of-sample-files-for-the-downstream-analysis-whywhy-not">Question 25: Can we use this output based on the limited number of sample files for the downstream analysis? Why/Why not?</h3>
<textarea id="question25" placeholder="Type your answer here..." class="input-box"></textarea>
</section>
<section id="question-26-what-does-it-mean-that-the-sample-tissues-have-been-fractionated-according-to-paper-please-describe-that-process-and-study-design.-will-more-fractions-lead-to-better-proteome-coverage" class="level3">
<h3 class="anchored" data-anchor-id="question-26-what-does-it-mean-that-the-sample-tissues-have-been-fractionated-according-to-paper-please-describe-that-process-and-study-design.-will-more-fractions-lead-to-better-proteome-coverage">Question 26: What does it mean that the sample tissues have been fractionated according to <em>PAPER</em>? Please describe that process and study design. Will more fractions lead to better proteome coverage?</h3>
<textarea id="question26" placeholder="Type your answer here..." class="input-box"></textarea>
<p>When the run in FragPipe is done, please locate the output results and get an overview of the output.</p>
</section>
<section id="question-27-what-kind-of-output-does-fragpipe-provide" class="level3">
<h3 class="anchored" data-anchor-id="question-27-what-kind-of-output-does-fragpipe-provide">Question 27: What kind of output does FragPipe provide?</h3>
<textarea id="question27" placeholder="Type your answer here..." class="input-box"></textarea>
<p>For the downstream analysis, we will use the output from the list of combined proteins, which we will explore further in the following section.</p>
</section>
</section>
<section id="further-interpretation-and-analysis-of-fragpipe-results" class="level2">
<h2 class="anchored" data-anchor-id="further-interpretation-and-analysis-of-fragpipe-results">Further Interpretation and Analysis of FragPipe Results</h2>
<p>For this part, we will use output files based on a run with FragPipe using all sample files (i.e., 5x72 raw files). That file can be downloaded here???</p>
<p>Now, we will look at the output from FragPipe, where we will use the file named <code>combined_proteins.tsv</code>. Initially, we will explore the contents of the file locally. Therefore, you should download the file from UCloud and view it locally in a file editor such as Excel.</p>
<p>You can download the file by clicking on the file in your output directory in the UCloud interface, from where you can choose to download it.</p>
<section id="question-28-give-a-brief-description-of-the-contents-of-the-table.-what-does-the-table-contain-across-the-rows-and-the-columns" class="level3">
<h3 class="anchored" data-anchor-id="question-28-give-a-brief-description-of-the-contents-of-the-table.-what-does-the-table-contain-across-the-rows-and-the-columns">Question 28: Give a brief description of the contents of the table. What does the table contain across the rows and the columns?</h3>
<textarea id="question28" placeholder="Type your answer here..." class="input-box"></textarea>
<p>For the downstream analysis, we will use the columns containing the TMT intensities across the proteins identified.</p>
<p>For that we will use <a href="https://computproteomics.bmb.sdu.dk/app_direct/OmicsQ/">OmicsQ</a>, which is a toolkit for quantitative proteomics. OmicsQ can be used to facilitate the processing of quantitative data from Omics type experiments. Additionally, it also serves as an entrypoint for using apps like <a href="https://computproteomics.bmb.sdu.dk/app_direct/PolySTest/">PolySTest</a> [SCHWAMMLE20201396] for statistical testing, <a href="https://computproteomics.bmb.sdu.dk/app_direct/VSClust/">VSClust</a> for clustering and <a href="https://computproteomics.bmb.sdu.dk/app_direct/ComplexBrowser/">ComplexBrowser</a> for the investigation of the behavior of protein complexes.</p>
</section>
<section id="reading-the-data" class="level3">
<h3 class="anchored" data-anchor-id="reading-the-data">Reading the Data</h3>
<!-- Placeholder for future content -->
<p>´´´r</p>
</section>
<section id="visualization-of-the-data" class="level3">
<h3 class="anchored" data-anchor-id="visualization-of-the-data">Visualization of the Data</h3>
<!-- Placeholder for future content -->
</section>
<section id="dynamic-range-similarity-and-missing-values" class="level3">
<h3 class="anchored" data-anchor-id="dynamic-range-similarity-and-missing-values">Dynamic Range, Similarity, and Missing Values</h3>
<!-- Placeholder for future content -->
</section>
<section id="principal-component-analysis" class="level3">
<h3 class="anchored" data-anchor-id="principal-component-analysis">Principal Component Analysis</h3>
<!-- Placeholder for future content -->
</section>
<section id="hierarchical-clustering" class="level3">
<h3 class="anchored" data-anchor-id="hierarchical-clustering">Hierarchical Clustering</h3>
<!-- Placeholder for future content -->
</section>
<section id="digging-deeper-into-the-behavior-of-one-protein" class="level3">
<h3 class="anchored" data-anchor-id="digging-deeper-into-the-behavior-of-one-protein">Digging Deeper into the Behavior of One Protein</h3>
<!-- Placeholder for future content -->
</section>
<section id="statistics-and-clustering-using-vsclust" class="level3">
<h3 class="anchored" data-anchor-id="statistics-and-clustering-using-vsclust">Statistics and Clustering using VSClust</h3>
<!-- Placeholder for future content -->
</section>
<section id="go-terms-and-pathways" class="level3">
<h3 class="anchored">GO Terms and Pathways</h3>
<!-- Placeholder for future content -->
<h2 class="anchored" data-anchor-id="go-terms-and-pathways">
Personal Details
</h2>
<div class="personal-info-grid">
<div class="personal-info-item">
<pre><code>&lt;label for="name"&gt;&lt;strong&gt;Name:&lt;/strong&gt;&lt;/label&gt;
&lt;textarea id="name" placeholder="Type your name here..." class="input-box"&gt;&lt;/textarea&gt;</code></pre>
</div>
<div class="personal-info-item">
<pre><code>&lt;label for="email"&gt;&lt;strong&gt;Email:&lt;/strong&gt;&lt;/label&gt;
&lt;textarea id="email" placeholder="Type your email here..." class="input-box"&gt;&lt;/textarea&gt;</code></pre>
</div>
<div class="personal-info-item">
<pre><code>&lt;label for="course"&gt;&lt;strong&gt;Course/Program:&lt;/strong&gt;&lt;/label&gt;
&lt;textarea id="course" placeholder="Type your course/program here..." class="input-box"&gt;&lt;/textarea&gt;</code></pre>
</div>
<div class="personal-info-item">
<pre><code>&lt;label for="date"&gt;&lt;strong&gt;Date:&lt;/strong&gt;&lt;/label&gt;
&lt;textarea id="date" placeholder="Type the date here..." class="input-box"&gt;&lt;/textarea&gt;</code></pre>
</div>
</div>
<style>
  /* Personal Information Grid Layout */
  .personal-info-grid {
    display: grid;
    grid-template-columns: repeat(2, 1fr); /* Two columns */
    gap: 20px; /* Space between columns */
    margin-bottom: 30px;
  }

  .personal-info-item {
    display: flex;
    flex-direction: column;
  }

  /* Styling the textarea */
  .input-box {
    width: 100%;
    min-height: 40px; /* Adjust height */
    padding: 10px;
    margin-top: 5px;
    margin-bottom: 15px;
    border: 1px solid #ccc;
    border-radius: 4px;
    font-family: Arial, sans-serif;
    font-size: 16px;
    background-color: #f9f9f9;
    resize: none;
    box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.1);
    transition: border-color 0.3s, box-shadow 0.3s;
  }

  .input-box:focus {
    border-color: #0073e6;
    box-shadow: 0px 4px 8px rgba(0, 115, 230, 0.2);
    outline: none;
  }
</style>
</section>
</section>
<section id="personal-details" class="level2">
<h2 class="anchored" data-anchor-id="personal-details">Personal Details</h2>
<strong>Name:</strong><br>

<textarea id="name" placeholder="Type your name here..." class="input-box"></textarea>
<strong>Email:</strong><br>

<textarea id="email" placeholder="Type your email here..." class="input-box"></textarea>
<strong>Course/Program:</strong><br>

<textarea id="course" placeholder="Type your course/program here..." class="input-box"></textarea>
<strong>Date:</strong><br>

<textarea id="date" placeholder="Type the date here..." class="input-box"></textarea>
<button id="downloadBtn" class="download-button">
Download Your Answers as PDF
</button>
<script src="https://cdnjs.cloudflare.com/ajax/libs/html2pdf.js/0.9.3/html2pdf.bundle.min.js"></script>
<script>
// Function to save answers to localStorage
function saveAnswers() {
  for (let i = 1; i <= 3; i++) { // Adjust based on the number of questions
    const answer = document.getElementById(`question${i}`)?.value || '';
    localStorage.setItem(`question${i}`, answer);
  }
}

// Load answers from localStorage
for (let i = 1; i <= 3; i++) {
  const savedAnswer = localStorage.getItem(`question${i}`) || '';
  if (document.getElementById(`question${i}`)) {
    document.getElementById(`question${i}`).value = savedAnswer;
  }
}

// Automatically resize textareas based on input
document.querySelectorAll('.input-box').forEach(textarea => {
  textarea.style.height = 'auto';
  textarea.style.height = (textarea.scrollHeight) + 'px';
  textarea.addEventListener('input', function() {
    this.style.height = 'auto';
    this.style.height = (this.scrollHeight) + 'px';
    saveAnswers();
  });
});

// Function to download the answers as a PDF with a more professional layout
document.getElementById('downloadBtn').addEventListener('click', function() {
  const answersHTML = `
    <div style="font-family: Arial, sans-serif; padding: 20px;">
      <h1 style="text-align: center; font-size: 24px;">Assignment Submission</h1>
      
      <h2 style="font-size: 18px; margin-bottom: 5px;">Question 1: Describe the central dogma of molecular biology</h2>
      <p style="font-size: 14px;">${document.getElementById('question1').value}</p>
      
      <h2 style="font-size: 18px; margin-bottom: 5px;">Question 2: Explain how mass spectrometry is used in proteomics</h2>
      <p style="font-size: 14px;">${document.getElementById('question2').value}</p>
      
      <h2 style="font-size: 18px; margin-bottom: 5px;">Question 3: Write a short code example in R or Python to visualize data</h2>
      <p style="font-size: 14px;">${document.getElementById('question3').value}</p>
      
      <!-- Add more questions and answers in the same format -->
    </div>
  `;

  const opt = {
    margin: 1,
    filename: 'assignment_answers.pdf',
    image: { type: 'jpeg', quality: 0.98 },
    html2canvas: { scale: 2 },
    jsPDF: { unit: 'in', format: 'letter', orientation: 'portrait' }
  };

  html2pdf().from(answersHTML).set(opt).save();
});
</script>
<style>
/* Styling the textarea */
.input-box {
  width: 100%;
  min-height: 120px;
  padding: 10px;
  margin-top: 10px;
  margin-bottom: 20px;
  border: 1px solid #ccc;
  border-radius: 4px;
  font-family: Arial, sans-serif;
  font-size: 16px;
  background-color: #f9f9f9;
  resize: none;
  box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.1);
  transition: border-color 0.3s, box-shadow 0.3s;
}

.input-box:focus {
  border-color: #0073e6;
  box-shadow: 0px 4px 8px rgba(0, 115, 230, 0.2);
  outline: none;
}

/* Styling the download button */
.download-button {
  background-color: #0073e6;
  color: white;
  padding: 10px 20px;
  border: none;
  border-radius: 4px;
  font-size: 16px;
  cursor: pointer;
  transition: background-color 0.3s, box-shadow 0.3s;
}

.download-button:hover {
  background-color: #005bb5;
  box-shadow: 0px 4px 8px rgba(0, 115, 230, 0.3);
}
</style>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>