---
title: Teaching Module
summary: Teaching module provided for the proteomics module. 
date: last-modified
hide:
  - navigation
---

<!--
# Put above to hide navigation (left), toc (right) or footer (bottom)

hide:
  - navigation 
  - toc
  - footer 

# You should hide the navigation if there are no subsections
# You should hide the Table of Contents if there are no important titles
-->

!!! note "Section Overview"

    &#128368; **Total Time Estimation:** 8 hours

    &#128172; **Learning Objectives:**    
        1. Understand the workflow of TMT-labeled mass spectrometry. <br>
        2. Retrieve and understand the study design and data from selected papers. <br>
        3. Download and preprocess proteomics data using FragPipe. <br>
        4. Set up and use FragPipe for TMT-labeled MS data analysis. <br>
        5. Create accurate annotation files for data analysis.
   


# Course Materials
This page provides access to course materials on proteomics. Please note that the page is currently under development, and more detailed course materials will be added in the future. 

This module provides a comprehensive guide to Tandem Mass Tag (TMT) labeled mass spectrometry data analysis. It is designed to equip students with the necessary skills to locate relevant scientific papers, retrieve and download proteomics data, process the data using FragPipe, and create appropriate annotation files for data analysis.

!!! note "Module Outline"

    1. Introduction to TMT Labeling and Mass Spectrometry <br>
    2. Retrieving Data and Study Design <br>
    3. Downloading and Processing Data using FragPipe <br>
    4. Setting Up Study Design in FragPipe <br>
    5. Creating Annotation Files for Each Pool <br>
    6. Data Analysis and Interpretation <br>
    7. Conclusion and Future Directions



## 1. Introduction to TMT Labeling and Mass Spectrometry

**Proteomics** is the large-scale study of proteins, particularly their structures and functions. As proteins are essential components of living organisms, understanding them is crucial for elucidating biological processes and disease mechanisms. **Mass-spectrometry (MS)-based proteomics** is a powerful technique that allows for the comprehensive analysis of proteins in a given sample.

## 2. Retrieving Data and Study Design
The first step in proteomics data analysis is to locate relevant scientific papers that describe the experimental design and data analysis methods. In this section, we will discuss how to retrieve proteomics data from the paper [Breast cancer quantitative proteome and proteogenomic landscape](https://www-nature-com.proxy1-bib.sdu.dk/articles/s41467-019-09018-y#Sec15) by Johansson et al. and understand the study design.


## 3. Downloading and Processing Data using FragPipe
In this section, we will download the raw proteomics data from the paper by Johansson et al. and process it using FragPipe, a popular proteomics data analysis software. We will discuss the steps involved in data processing and the parameters used in the analysis.

## 4. Setting Up Study Design in FragPipe
To analyze the TMT-labeled mass spectrometry data, we need to set up the study design in FragPipe. This involves creating annotation files for each pool of samples and specifying the experimental conditions. In this section, we will discuss how to set up the study design in FragPipe.

## 5. Creating Annotation Files for Each Pool
The annotation files are essential for data analysis in FragPipe. They contain information about the experimental conditions, sample names, and other relevant details. In this section, we will discuss how to create accurate annotation files for each pool of samples.

## 6. Data Analysis and Interpretation
Data analysis is a crucial step in proteomics research. In this section, we will analyze the TMT-labeled mass spectrometry data using FragPipe and interpret the results. We will discuss how to identify differentially expressed proteins and pathways that are associated with breast cancer.

## 7. Conclusion and Future Directions
In this final section, we will summarize the key findings from the data analysis and discuss future directions for proteomics research. We will also reflect on the challenges and opportunities in the field of proteomics and the potential impact of proteomics on human health.

# References