<!doctype html><html lang=en class=no-js> <head><script type=application/ld+json>
    {
      "@context": "https://schema.org",
      "@id": "https://hds-sandbox.github.io/proteomics-sandbox/",
      "@type": "LearningResource",
      "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
      "description": "A resource for biomedical students and researchers. Includes proteomics software tools like FragPipe, MaxQuant, PDV, SearchGUI, ThermoRawFileParser, and PeptideShaker. Offers a user-friendly interface, automated identification and quantification, comprehensive data analysis, and lightweight clone feature for optimized storage.",
      "keywords": [
        "proteomics, bioinformatics"
      ],
      "learningResourceType": [
        "e-Learning"
      ],
      "license": [
        {
          "@id": "https://creativecommons.org/licenses/by-sa/4.0/",
          "@type": "CreativeWork"
        }
      ],
      "mentions": [
        {
          "@id": "# Link to DOI or zenodo materials (10.5281/zenodo.7660225)",
          "@type": "Thing"
        },
        {
          "@id": "#Another link to DOI or zenodo materials (10.5281/zenodo.7565997)",
          "@type": "Thing"
        }
      ],
      "name" : "Proteomics Sandbox",
      "educationallevel" : "Beginner",
      "audience" : "MSc, PhD"
    }
    </script><title>Teaching Module - Clinical Proteomics</title><link rel=stylesheet href=assets/stylesheets/main.c4a75a56.min.css><link rel=stylesheet href=assets/stylesheets/palette.a0c5b2b5.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=stylesheets/neoteroi-mkdocs.css><link rel=stylesheet href=stylesheets/extra.css><script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script>"undefined"!=typeof __md_analytics&&__md_analytics()</script></head> <body dir=ltr data-md-color-scheme=brightness data-md-color-primary data-md-color-accent> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#course-materials class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class="md-header md-header--shadow md-header--lifted" data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=http://hds-sandbox.github.io/ title="Clinical Proteomics" class="md-header__button md-logo" aria-label="Clinical Proteomics" data-md-component=logo> <img src=images/logo.png alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Clinical Proteomics </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Teaching Module </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=brightness data-md-color-primary data-md-color-accent aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary data-md-color-accent aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" title=Share aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg> </a> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/hds-sandbox/proteomics-sandbox title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> proteomics-course </div> </a> </div> </nav> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=index.html class=md-tabs__link> Introduction </a> </li> <li class=md-tabs__item> <a href=gettingstarted.html class=md-tabs__link> Getting Started </a> </li> <li class=md-tabs__item> <a href=setup.html class=md-tabs__link> Setup </a> </li> <li class=md-tabs__item> <a href=colabfold.html class=md-tabs__link> ColabFold </a> </li> <li class=md-tabs__item> <a href=coursematerials.html class=md-tabs__link> Course Materials </a> </li> <li class=md-tabs__item> <a href=teachingmodule.html class="md-tabs__link md-tabs__link--active"> Teaching Modules </a> </li> <li class=md-tabs__item> <a href=contributors.html class=md-tabs__link> Contributors </a> </li> </ul> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation hidden> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=http://hds-sandbox.github.io/ title="Clinical Proteomics" class="md-nav__button md-logo" aria-label="Clinical Proteomics" data-md-component=logo> <img src=images/logo.png alt=logo> </a> Clinical Proteomics </label> <div class=md-nav__source> <a href=https://github.com/hds-sandbox/proteomics-sandbox title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> proteomics-course </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=index.html class=md-nav__link> Introduction </a> </li> <li class=md-nav__item> <a href=gettingstarted.html class=md-nav__link> Getting Started </a> </li> <li class=md-nav__item> <a href=setup.html class=md-nav__link> Setup </a> </li> <li class=md-nav__item> <a href=colabfold.html class=md-nav__link> ColabFold </a> </li> <li class=md-nav__item> <a href=coursematerials.html class=md-nav__link> Course Materials </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <a href=teachingmodule.html class="md-nav__link md-nav__link--active"> Teaching Modules </a> </li> <li class=md-nav__item> <a href=contributors.html class=md-nav__link> Contributors </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <div><h1 id=course-materials>Course Materials<a class=headerlink href=#course-materials title="Permanent link">¶</a></h1> <p>This page provides access to course materials on proteomics. Please note that the page is currently under development, and more detailed course materials will be added in the future. </p> <h1 id=introduction-to-mass-spectrometry-based-proteomics>Introduction to Mass-Spectrometry Based Proteomics<a class=headerlink href=#introduction-to-mass-spectrometry-based-proteomics title="Permanent link">¶</a></h1> <p><strong>Proteomics</strong> is the large-scale study of proteins, particularly their structures and functions. As proteins are essential components of living organisms, understanding them is crucial for elucidating biological processes and disease mechanisms. <strong>Mass-spectrometry (MS)-based proteomics</strong> is a powerful technique that allows for the comprehensive analysis of proteins in a given sample.</p> <h2 id=overview-of-proteomics-data-workflow>Overview of Proteomics Data Workflow<a class=headerlink href=#overview-of-proteomics-data-workflow title="Permanent link">¶</a></h2> <p>Understanding the workflow of MS-based proteomics is crucial for effective data analysis. Below is a step-by-step guide:</p> <ol> <li><strong>Sample Preparation</strong>: </li> <li>Protein extraction and digestion into peptides using enzymes like trypsin.</li> <li> <p>Peptide purification and concentration.</p> </li> <li> <p><strong>Liquid Chromatography-Mass Spectrometry (LC-MS)</strong>:</p> </li> <li>Separation of peptides by liquid chromatography.</li> <li> <p>Ionization of peptides and analysis by mass spectrometry.</p> </li> <li> <p><strong>Data Acquisition</strong>:</p> </li> <li> <p>Collection of MS spectra representing peptide masses and their fragments.</p> </li> <li> <p><strong>Database Search</strong>:</p> </li> <li>Matching MS spectra against protein databases using search engines.</li> <li> <p>Generating peptide-spectrum matches (PSMs).</p> </li> <li> <p><strong>Data Validation</strong>:</p> </li> <li>Filtering and validating identified peptides and proteins.</li> <li> <p>Estimating false discovery rates (FDR).</p> </li> <li> <p><strong>Quantitative Analysis</strong>:</p> </li> <li>Quantifying protein abundance across different samples.</li> <li> <p>Analyzing differential expression and PTMs.</p> </li> <li> <p><strong>Data Interpretation</strong>:</p> </li> <li>Biological interpretation of the proteomics data.</li> <li>Integration with other omics data.</li> </ol> <p><img alt="Proteomics Workflow" src=path/to/your/image.png></p> <h3 id=questions>Questions<a class=headerlink href=#questions title="Permanent link">¶</a></h3> <details> <summary>What are the main steps in the proteomics data workflow?</summary> 1. Sample Preparation 2. Liquid Chromatography-Mass Spectrometry (LC-MS) 3. Data Acquisition 4. Database Search 5. Data Validation 6. Quantitative Analysis 7. Data Interpretation </details> <h2 id=types-of-mass-spectrometers>Types of Mass Spectrometers<a class=headerlink href=#types-of-mass-spectrometers title="Permanent link">¶</a></h2> <p>Different types of mass spectrometers are used in proteomics, each with specific advantages:</p> <ul> <li><strong>MALDI-TOF</strong>: Matrix-Assisted Laser Desorption/Ionization-Time of Flight, suitable for large biomolecules.</li> <li><strong>ESI-QTOF</strong>: Electrospray Ionization-Quadrupole Time of Flight, known for high resolution and accuracy.</li> <li><strong>Orbitrap</strong>: Offers high resolution and mass accuracy, widely used in proteomics.</li> </ul> <h3 id=questions_1>Questions<a class=headerlink href=#questions_1 title="Permanent link">¶</a></h3> <details> <summary>What are the different types of mass spectrometers used in proteomics?</summary> - MALDI-TOF - ESI-QTOF - Orbitrap </details> <h2 id=data-generation-in-proteomics>Data Generation in Proteomics<a class=headerlink href=#data-generation-in-proteomics title="Permanent link">¶</a></h2> <p>The process starts with the extraction of proteins from a biological sample. These proteins are then digested into smaller peptides, usually using an enzyme like trypsin. The resulting peptides are separated by liquid chromatography and then ionized before being introduced into a mass spectrometer. The mass spectrometer measures the mass-to-charge ratio of the peptides, generating spectra that represent the peptide’s mass and its fragments. These spectra are then used to infer the peptide's sequence and, ultimately, the identity and quantity of the proteins in the sample.</p> <h3 id=questions_2>Questions<a class=headerlink href=#questions_2 title="Permanent link">¶</a></h3> <details> <summary>What are the steps involved in generating data in proteomics?</summary> 1. Protein extraction from the biological sample 2. Digestion of proteins into peptides 3. Separation of peptides by liquid chromatography 4. Ionization and mass spectrometry analysis 5. Generation of spectra representing peptide masses and fragments 6. Inference of peptide sequence, identity, and quantity of proteins </details> <h2 id=importance-and-applications>Importance and Applications<a class=headerlink href=#importance-and-applications title="Permanent link">¶</a></h2> <p>Proteomics is vital for understanding cellular processes, identifying biomarkers for diseases, and discovering new drug targets. It can be used in various fields such as cancer research, neurobiology, and infectious disease studies. By analyzing the protein expression profiles of different conditions, researchers can gain insights into disease mechanisms and identify potential therapeutic targets.</p> <h3 id=questions_3>Questions<a class=headerlink href=#questions_3 title="Permanent link">¶</a></h3> <details> <summary>Why is proteomics important and what are its applications?</summary> Proteomics is important for: - Understanding cellular processes - Identifying biomarkers for diseases - Discovering new drug targets Applications include: - Cancer research - Neurobiology - Infectious disease studies </details> <h1 id=bioinformatics-analysis-in-proteomics>Bioinformatics Analysis in Proteomics<a class=headerlink href=#bioinformatics-analysis-in-proteomics title="Permanent link">¶</a></h1> <p>The vast amount of data generated by MS-based proteomics requires sophisticated bioinformatics tools for analysis. These tools help to identify and quantify proteins, analyze post-translational modifications, and compare protein expression levels across different samples.</p> <h2 id=database-search-and-spectral-libraries>Database Search and Spectral Libraries<a class=headerlink href=#database-search-and-spectral-libraries title="Permanent link">¶</a></h2> <p>Protein identification involves matching MS spectra against protein databases:</p> <ul> <li><strong>Database Search</strong>: Algorithms like Mascot and MS-GF+ search for peptide sequences that match the observed spectra.</li> <li><strong>Spectral Libraries</strong>: Collections of known spectra used to identify peptides with high confidence.</li> </ul> <h3 id=questions_4>Questions<a class=headerlink href=#questions_4 title="Permanent link">¶</a></h3> <details> <summary>What are the two main methods for protein identification?</summary> - Database Search - Spectral Libraries </details> <h2 id=data-quality-and-validation>Data Quality and Validation<a class=headerlink href=#data-quality-and-validation title="Permanent link">¶</a></h2> <p>Ensuring data quality and validation is critical:</p> <ul> <li><strong>Quality Control</strong>: Implementing standards and replicates.</li> <li><strong>Validation</strong>: Using false discovery rates (FDR) and decoy databases to confirm peptide and protein identifications.</li> </ul> <h3 id=questions_5>Questions<a class=headerlink href=#questions_5 title="Permanent link">¶</a></h3> <details> <summary>How is data quality ensured in proteomics?</summary> - Implementing standards and replicates for quality control - Using false discovery rates (FDR) and decoy databases for validation </details> <h2 id=quantitative-proteomics>Quantitative Proteomics<a class=headerlink href=#quantitative-proteomics title="Permanent link">¶</a></h2> <p>Quantitative approaches in proteomics:</p> <ul> <li><strong>Label-Free Quantification</strong>: Measures peptide intensities directly from MS data.</li> <li><strong>SILAC (Stable Isotope Labeling by Amino acids in Cell culture)</strong>: Incorporates stable isotopes into proteins for relative quantification.</li> <li><strong>TMT (Tandem Mass Tags)</strong>: Uses chemical tags for multiplexed quantification.</li> </ul> <h3 id=questions_6>Questions<a class=headerlink href=#questions_6 title="Permanent link">¶</a></h3> <details> <summary>What are the main quantitative approaches in proteomics?</summary> - Label-Free Quantification - SILAC (Stable Isotope Labeling by Amino acids in Cell culture) - TMT (Tandem Mass Tags) </details> <h2 id=post-translational-modifications-ptms>Post-Translational Modifications (PTMs)<a class=headerlink href=#post-translational-modifications-ptms title="Permanent link">¶</a></h2> <p>PTMs are critical for protein function:</p> <ul> <li><strong>Types of PTMs</strong>: Phosphorylation, glycosylation, ubiquitination, etc.</li> <li><strong>Analysis Tools</strong>: Software for detecting and quantifying PTMs.</li> </ul> <h3 id=questions_7>Questions<a class=headerlink href=#questions_7 title="Permanent link">¶</a></h3> <details> <summary>What are some common post-translational modifications (PTMs)?</summary> - Phosphorylation - Glycosylation - Ubiquitination </details> <h2 id=challenges-and-limitations>Challenges and Limitations<a class=headerlink href=#challenges-and-limitations title="Permanent link">¶</a></h2> <p>Common challenges in proteomics include:</p> <ul> <li><strong>Sample Complexity</strong>: High complexity and dynamic range of proteomes.</li> <li><strong>Data Interpretation</strong>: Integrating large datasets and biological relevance.</li> </ul> <h3 id=questions_8>Questions<a class=headerlink href=#questions_8 title="Permanent link">¶</a></h3> <details> <summary>What are common challenges in proteomics?</summary> - Sample complexity - Data interpretation </details> <h2 id=software-tools>Software Tools<a class=headerlink href=#software-tools title="Permanent link">¶</a></h2> <p>Several software tools are used in proteomics data analysis, including Mascot, X! Tandem, and MaxQuant. However, this teaching material will focus on <strong>SearchGUI</strong> and <strong>PeptideShaker</strong>.</p> <h3 id=searchgui>SearchGUI<a class=headerlink href=#searchgui title="Permanent link">¶</a></h3> <p>SearchGUI is an open-source graphical user interface that allows users to configure and run proteomics search engines like X! Tandem, MS-GF+, and Comet. It provides a user-friendly platform for setting up the search parameters, selecting the appropriate databases, and running multiple search engines simultaneously. This enhances the reliability of peptide and protein identification by combining results from different algorithms.</p> <h3 id=peptideshaker>PeptideShaker<a class=headerlink href=#peptideshaker title="Permanent link">¶</a></h3> <p>PeptideShaker is a comprehensive platform for the visualization and analysis of proteomics data. It takes the search results generated by SearchGUI and integrates them into an intuitive interface. PeptideShaker allows users to validate and interpret the results, perform protein inference, and quantify proteins across different samples. It also provides various statistical and visualization tools to help users understand their data better.</p> <h3 id=questions_9>Questions<a class=headerlink href=#questions_9 title="Permanent link">¶</a></h3> <details> <summary>What are the functions of SearchGUI and PeptideShaker?</summary> - **SearchGUI**: Configures and runs proteomics search engines, enhances reliability by combining results from different algorithms. - **PeptideShaker**: Visualizes and analyzes proteomics data, validates and interprets results, performs protein inference, quantifies proteins, and provides statistical and visualization tools. </details> <h2 id=usage-and-benefits>Usage and Benefits<a class=headerlink href=#usage-and-benefits title="Permanent link">¶</a></h2> <p>SearchGUI and PeptideShaker streamline the proteomics workflow by automating complex steps and providing powerful tools for data analysis. They are widely used due to their flexibility, compatibility with various search engines, and robust visualization capabilities. These tools enable researchers to handle large datasets efficiently and derive meaningful biological insights, making them indispensable in modern proteomics research.</p> <h3 id=questions_10>Questions<a class=headerlink href=#questions_10 title="Permanent link">¶</a></h3> <details> <summary>Why are SearchGUI and PeptideShaker widely used in proteomics?</summary> - They streamline the proteomics workflow - Provide powerful tools for data analysis - Offer flexibility and compatibility with various search engines - Provide robust visualization capabilities </details> <h1 id=case-studies-and-applications>Case Studies and Applications<a class=headerlink href=#case-studies-and-applications title="Permanent link">¶</a></h1> <p>Examples of impactful proteomics studies:</p> <ul> <li><strong>Cancer Research</strong>: Identification of biomarkers for early detection.</li> <li><strong>Neurobiology</strong>: Understanding protein changes in neurodegenerative diseases.</li> </ul> <h1 id=future-trends-in-proteomics-and-bioinformatics>Future Trends in Proteomics and Bioinformatics<a class=headerlink href=#future-trends-in-proteomics-and-bioinformatics title="Permanent link">¶</a></h1> <p>Emerging trends and technologies:</p> <ul> <li><strong>Artificial Intelligence</strong>: Machine learning algorithms for data analysis.</li> <li><strong>Single-Cell Proteomics</strong>: Techniques for analyzing proteins at the single-cell level.</li> </ul> <h1 id=resources-and-further-reading>Resources and Further Reading<a class=headerlink href=#resources-and-further-reading title="Permanent link">¶</a></h1> <p>For further learning, explore the following resources:</p> <ul> <li><strong>Textbooks</strong>: "Proteomics: Principles, Techniques, and Applications" by Richard Twyman.</li> <li><strong>Online Courses</strong>: Coursera, edX, and other platforms offer courses on proteomics.</li> <li><strong>Workshops</strong>: Attend workshops and conferences to stay updated on the latest developments.</li> </ul></div> </article> </div> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </button> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=coursematerials.html class="md-footer__link md-footer__link--prev" aria-label="Previous: Course Materials" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> Course Materials </div> </div> </a> <a href=contributors.html class="md-footer__link md-footer__link--next" aria-label="Next: Contributors" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> Contributors </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=https://github.com/hds-sandbox target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://twitter.com/ucph_heads target=_blank rel=noopener title=twitter.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 6.3.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": ".", "features": ["content.code.annotate", "content.tooltips", "navigation.indexes", "navigation.sections", "navigation.footer", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow"], "search": "assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script> <script src=assets/javascripts/bundle.efa0ade1.min.js></script> <script src=javascripts/mathjax.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>